src/cls/IPM/Lifecycle/Base.cls

```diff
@@ -890,7 +894,11 @@ Method %Compile(ByRef pParams) As %Status
 			Quit:tKey=""
 			
 			If '$Data(tHandledMap(tResource.Name)) && $IsObject(tResource.Processor) {
-				If tResource.Processor.%IsA("%IPM.ResourceProcessor.AbstractCompilable") {
+				If tResource.Processor.%IsA("%IPM.ResourceProcessor.AbstractCompilable") && tResource.Processor.CompileFromProject {
+					If tVerbose {
+						Write !, tResource.Name_" was loaded from studio project, skipping compilation."
```

Comment:
Nit: rather than saying studio-project here, perhaps says "was loaded from deployed code file" since the format of deployed code is irrelevant. Also, instead of "skipping compilation", maybe "so compilation isn't needed"? cc @isc-tleavitt thoughts on wording?

Reply:
Let's use wording:
tResource.Name_" was loaded as deployed code, so compilation isn't needed."